src/forms/FormAutocomplete.jsx

```diff
@@ -105,7 +113,7 @@ export function FormAutocomplete({
       setValue('');
       setSelectedItem(null);
       updateSearchInputValidation();
-    } else if (isEmptyLike(selectedItem) && !isEmptyLike(searchValue) && allowUnlistedValue) {
+    } else if (selectedItem?.value !== searchValue && !isEmptyLike(searchValue) && allowUnlistedValue) {
```

Comment:
Se a validação for `isEmptyLike(selectedItem)`, o usuário não conseguia colocar um "unlisted value" se o campo já estivesse preenchido. Ele teria que sempre limpar o campo e aí sim preencher com um valor "unlisted".